src/3p.js

```diff
@@ -171,3 +171,18 @@ export function validateData(data, allowedFields) {
         'Unknown attribute for %s: %s.', data.type, field);
   }
 }
+
+/**
+ * Function to check if data has all mandatory
+ * fields specified
+ * @param {!Object} data
+ * @param {!Array<string>} fields
+ */
+export function dataHasAllFields(data, fields) {
+  for (let c = 0; c < fields.length; c++) {
+    if (!data.hasOwnProperty(fields[c]) && !data[fields[c]]) {
```

Comment:
I think truthiness might be a bit too much for the general case. How about you do extended validation in the script you load?